vic/drivers/classic/src/alloc_veg_hist.c

```diff
@@ -38,29 +38,27 @@ alloc_veg_hist(int                nrecs,
 {
     int i, j;
 
-    (*veg_hist) = (veg_hist_struct **) calloc(nrecs, sizeof(veg_hist_struct *));
+    (*veg_hist) = calloc(nrecs, sizeof(*(*veg_hist)));
     if ((*veg_hist) == NULL) {
         log_err("Memory allocation error in alloc_veg_hist().");
     }
 
     for (i = 0; i < nrecs; i++) {
-        (*veg_hist)[i] = (veg_hist_struct *) calloc(nveg,
-                                                    sizeof(veg_hist_struct));
+        (*veg_hist)[i] = calloc(nveg, sizeof(*(*veg_hist)[i]));
```

Comment:
So, how is this related to the port of veg_hist that you wanted me to do?
I'm still hoping to do that - and soon, now that I got past the urgent
paper I had to write.  Let me know how I can help.

Ted

On Fri, Aug 28, 2015 at 5:25 PM, Bart Nijssen <notifications@github.com>
wrote:

> In vic/drivers/classic/src/alloc_veg_hist.c
> <https://github.com/UW-Hydro/VIC/pull/257#discussion_r38256094>:
>
> >      if ((*veg_hist) == NULL) {
> >          log_err("Memory allocation error in alloc_veg_hist().");
> >      }
> >
> >      for (i = 0; i < nrecs; i++) {
> > -        (*veg_hist)[i] = (veg_hist_struct *) calloc(nveg,
> > -                                                    sizeof(veg_hist_struct));
> > +        (*veg_hist)[i] = calloc(nveg, sizeof(*(*veg_hist)[i]));
>
> Note that a compiler would be unlikely to complain about either, since it
> will let you allocate any amount of memory that you want, even if it is the
> incorrect amount
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/UW-Hydro/VIC/pull/257/files#r38256094>.
>